defaults/main.yml

```diff
@@ -2,7 +2,9 @@
 ## general
 spamassassin_user: spamd
 spamassassin_group: spamd
-spamassassin_home_dir: /var/log/spamassassin/
+spamassassin_home_dir: /var/lib/spamassassin/
+spamassassin_log_dir: /var/log/
+spamassassin_log_file: spamd.log
```

Comment:
These variables aren't used anywhere

Reply:
Sorry - I forgot the changed default template. I just wanted to have the spamd.log in /var/log/, that is not possible in the current version.

Reply:
That is fixed now.

Next message will explain the "why"

Reply:
Something happened on Buster - It worked fine using Stretch.

On a clean system: apt install 'spamassassin', 'spamc', 'libmail-dkim-perl' will start these processes as well:

debian-+ 15439     1  0 20:24 ?        00:00:00 /lib/systemd/systemd --user
debian-+ 15440 15439  0 20:24 ?        00:00:00 (sd-pam)
debian-+ 15461     1  0 20:24 ?        00:00:00 gpg-agent --homedir /var/lib/spamassassin/sa-update-keys --use-standard-socket --daemon

Note that they all have debian-spamd as the owning user. I have no clue why they are started. The weird thing is that spamassassin itself is not started.

The result is that "ensure spamd user is present" will fail since the user "debian-spamd" is in use.

Fix - first try, strategy: move the "ensure spamd user is present" before spamassassin install
This worked, until I tried to rerun the playbook on the same system I had previously installed on (idempotency check). This failed since the processes above were still running + spamassassin was also running.

Second try: strategy: stop spamassassin, kill processes owned by spamd, proceed as in first try.
This worked - almost. IIRC the install fails because spamd-user has "/bin/false" as shell.

Third try: also change spamd user to have shell "/bin/sh", then proceed as in second try.

Success!

However, the code is not as straight forward and clean as before.

This was my experience with Buster, YMMV